madprops/serializers.py

```diff
@@ -83,12 +86,25 @@ class Property(model):
     }
 
     And an input dictionary of the above structure will be converted to
-    the collection of Property instances
+    the collection of Property instances.
     """
 
     _options_class = PropertySerializerOptions
 
+    def __new__(cls, *args, **kwargs):
+        if (kwargs.pop('many', False) or len(kwargs.get('data', [])) > 1):
+            # TODO: need to review this in future. Didn't find a better way
+            # to prevent recursion here.
+            if not kwargs.get('called_from_list_serializer'):
+                kwargs['called_from_list_serializer'] = 1
+                return cls.many_init(*args, **kwargs)
+        return super(BaseSerializer, cls).__new__(cls, *args, **kwargs)
+
     def __init__(self, *args, **kwargs):
+        # TODO: need to review this in future. Didn't find a better way
+        # to prevent recursion here.
```

Comment:
Probably we shouldn't mention the same TODO twice - the one in the `__new__` will be enough